core/src/test/scala/org/apache/spark/scheduler/BlacklistTrackerSuite.scala

```diff
@@ -480,17 +479,13 @@ class BlacklistTrackerSuite extends SparkFunSuite with BeforeAndAfterEach with M
   test("blacklisting kills executors, configured by BLACKLIST_KILL_ENABLED") {
     val allocationClientMock = mock[ExecutorAllocationClient]
     when(allocationClientMock.killExecutors(any(), any(), any(), any())).thenReturn(Seq("called"))
-    when(allocationClientMock.killExecutorsOnHost("hostA")).thenAnswer(new Answer[Boolean] {
-      // To avoid a race between blacklisting and killing, it is important that the nodeBlacklist
-      // is updated before we ask the executor allocation client to kill all the executors
-      // on a particular host.
-      override def answer(invocation: InvocationOnMock): Boolean = {
-        if (blacklist.nodeBlacklist.contains("hostA") == false) {
-          throw new IllegalStateException("hostA should be on the blacklist")
-        }
+    when(allocationClientMock.killExecutorsOnHost("hostA")).thenAnswer((_: InvocationOnMock) =>
+      if (blacklist.nodeBlacklist.contains("hostA")) {
```

Comment:
Is the comment lost by purpose?

Reply:
Oops, no probably lost by automated refactoring

Reply:
I am kinda glad. Because I started to question whether I would like to read this all... Now I see it has value so, I continue it :)